scripts/missions/amk/13_A_Challenge_You_Could_Be_a_Winner.lua

```diff
@@ -17,28 +19,353 @@ mission.reward =
 
 mission.sections =
 {
+    -- Part 1: Castle Zvahl Baileys
     {
         check = function(player, currentMission, missionStatus, vars)
-            return currentMission == mission.missionId
+            return currentMission >= mission.missionId and
+            not player:hasKeyItem(xi.ki.GAUNTLET_CHALLENGE_KUPON)
+        end,
+
+        [xi.zone.CASTLE_ZVAHL_BAILEYS] =
+        {
+            ['Shadowy_Pillar'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(100, 3)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [100] = function(player, csid, option, npc)
+                    player:setCharVar('Mission[10][12]progress', 1)
```

Comment:
Don't you just call `mission.setVar('progress')` or something here, rather than making up the key by hand?

Reply:
Same question for all manual management and lookup of these keys

Reply:
Actually scratch that, this mission is run while on the following mission.  So I'm not sure how it would work using the mission progress variable while on a different mission in case that one needed to use it. Thoughts?

Reply:
That's fine then, just leave a note saying that's whats happening